crates/q_cli/src/cli/chat/tools/use_aws.rs

```diff
@@ -0,0 +1,237 @@
+use std::collections::HashMap;
+use std::io::Write;
+use std::process::Stdio;
+
+use bstr::ByteSlice;
+use crossterm::{
+    queue,
+    style,
+};
+use eyre::{
+    Result,
+    WrapErr,
+};
+use fig_os_shim::Context;
+use serde::Deserialize;
+
+use super::{
+    InvokeOutput,
+    OutputKind,
+};
+
+const ALLOWED_OPS: [&str; 6] = ["get", "describe", "list", "ls", "search", "batch_get"];
+
+#[derive(Debug, thiserror::Error)]
+enum AwsToolError {
+    ForbiddenOperation(String),
+}
+
+impl std::fmt::Display for AwsToolError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AwsToolError::ForbiddenOperation(op) => Ok(writeln!(f, "Forbidden operation encountered: {}", op)?),
+        }
+    }
+}
+
+// TODO: we should perhaps composite this struct with an interface that we can use to mock the
+// actual cli with. That will allow us to more thoroughly test it.
+#[derive(Debug, Deserialize)]
+pub struct UseAws {
+    pub service_name: String,
+    pub operation_name: String,
+    pub parameters: HashMap<String, String>,
+    pub region: String,
+    pub profile_name: Option<String>,
+    pub label: Option<String>,
+}
+
+impl UseAws {
+    fn validate_operation(&self) -> Result<(), AwsToolError> {
+        let operation_name = &self.operation_name;
+        for op in ALLOWED_OPS {
+            if self.operation_name.starts_with(op) {
+                return Ok(());
+            }
+        }
+        Err(AwsToolError::ForbiddenOperation(operation_name.clone()))
+    }
+
+    pub fn display_name() -> String {
+        "Use AWS".to_owned()
+    }
+
+    pub async fn invoke(&self, _ctx: &Context, _updates: impl Write) -> Result<InvokeOutput> {
+        let mut command = tokio::process::Command::new("aws");
+        let profile_name = if let Some(ref profile_name) = self.profile_name {
```

Comment:
Should you pass any profile if one isnt specified? You may want to inherit from the AWS_PROFILE env var in which case this would not be desired.